integration_tests/src/main/python/data_gen.py

```diff
@@ -208,6 +209,37 @@ def __init__(self, nullable=True, min_val = INT_MIN, max_val = INT_MAX,
     def start(self, rand):
         self._start(rand, lambda : rand.randint(self._min_val, self._max_val))
 
+
+#Maximum number of decimal digits a Long can represent is 18.
+DECIMAL_MIN = -(1 << 59)
+DECIMAL_MAX = (1 << 59)  #approximate
```

Comment:
I don't think these work the way we want them to because the maximum and minimum values are dependent on the precision and scale.  These values are invalid for some combinations of these.

I would really prefer to have us generate a string that represents the correct values, using something like the pattern below.

```
Decimal(('9'*precision) + 'e' + str(-scale)), Decimal('-' + ('9' * precision) + 'e' + str(-scale))
```

Reply:
done